src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import ModalCloseIcon from "@/icons/ModalCloseIcon";
+import { useEffect } from "react";
+
+type ModalProps = {
+  isModalOpen: boolean;
+  setIsModalOpen: (isOpen: boolean) => void;
+  children: React.ReactNode;
+};
+
+const Modal = ({ children, isModalOpen, setIsModalOpen }: ModalProps) => {
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setIsModalOpen(false);
+      }
+    };
+    window.addEventListener("keydown", handleKeyDown);
+    return () => window.removeEventListener("keydown", handleKeyDown);
+  }, [setIsModalOpen]);
+
+  return (
+    isModalOpen && (
+      <div
+        onClick={() => setIsModalOpen(false)}
```

Comment:
Clicking on the modal's backdrop closes the modal. However, this will also close the modal if the user clicks inside the modal itself.

Reply:
I've added `onClick={(e) => e.stopPropagation()}` in child component to prevent closing modal by clicking inside modal itself